### PR TITLE
Fix Windows build warnings in template and core tests

### PR DESCRIPTION
**Pull request by panhaiqi**
### Details:
 - *Fix Windows build warnings in template and core tests*

### Tickets:
 - *103761*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 1]
```diff
@@ -936,7 +936,9 @@ void program_node::init_onednn_primitive_attributes() {
             } else if (fused_desc->activation_function == cldnn::activation_func::hsigmoid) {
                 // hard_sigmoid(x,a,b) = clamp(ax+b, 0, 1)
                 // hsigmoid(x) = clamp(val+3, 0, 6) / 6 = clamp(val/6+0.5, 0, 1) = hard_sigmoid(val, 1/6, 1/2)
-                post_ops.append_eltwise(dnnl::algorithm::eltwise_hardsigmoid, 1./6, 1./2);
+                post_ops.append_eltwise(dnnl::algorithm::eltwise_hardsigmoid,
+                                        static_cast<float>(1. / 6),
+                                        static_cast<float>(1. / 2));
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/graph/program_node.cpp:941]: why not 1.0f ?

Comment (ilyachur) [reply src/plugins/intel_gpu/src/graph/program_node.cpp:941]: @panhaiqi Please take a look, it should help to avoid static_cast.

**Comment by ilyachur** [comment]
Please fix CI

File: src/core/tests/type_prop/matmul.cpp [hunk 2]
```diff
@@ -552,6 +552,6 @@ TEST(type_prop, matmul_propagate_label_on_b_input_after_reshape) {
     const auto& output_shape = matmul->get_output_partial_shape(0);
     const auto labels = get_shape_labels(output_shape);
 
-    ASSERT_THAT(labels, ElementsAre(my_label, 0, 0));
+    ASSERT_THAT(labels, ElementsAre((int)my_label, 0, 0));
```
Comment (praasz) [inline src/core/tests/type_prop/matmul.cpp:555]: This cast should not be here.
The `my_label` should have correct type.
Change from `size_t` -> `ov::label_t`

File: src/plugins/template/tests/functional/op_reference/softmax.cpp [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 using namespace reference_tests;
 using namespace ov;
-
+#pragma warning(disable : 4756)
```
Comment (ilyachur) [inline src/plugins/template/tests/functional/op_reference/softmax.cpp:12]: Why do you suppress this warning?
Also we should disable it only for windows

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**